packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -149,26 +149,53 @@ class InnerBlocks extends Component {
 			clientId,
 			hasOverlay,
 			__experimentalCaptureToolbars: captureToolbars,
+			forwardedRef,
 			...props
 		} = this.props;
 		const { templateInProcess } = this.state;
 
+		if ( templateInProcess ) {
+			return null;
+		}
+
+		const blockList = (
+			<BlockList
+				rootClientId={ clientId }
+				forwardedRef={ forwardedRef }
+				{ ...omit( props, [
+					'templateLock',
+					'isSmallScreen',
+					'block',
+					'blockListSettings',
+					'parentLock',
+					'isLastBlockChangePersistent',
+					'replaceInnerBlocks',
+					'__unstableMarkNextChangeAsNotPersistent',
+					'updateNestedSettings',
+					'__experimentalMoverDirection',
+					'allowedBlocks',
+					'template',
+				] ) }
+			/>
+		);
+		if ( props.tagName ) {
+			return blockList;
```

Comment:
What about `has-overlay`, `is-capturing-toolbar` and `block-editor-inner-blocks` classNames, don't we need them when we provide a tagName?

Also why the extra "div" when we don't provide a tagName? Do you think it can just be a fallback value for "tagName"?

Reply:
Yes, either we should merge #19849 first, or just incorporate it into this PR.

> Also why the extra "div" when we don't provide a tagName?

For back compat... I don't want to invalidate styling for existing blocks using `InnerBlocks` without a tag name.